src/main/scala/org/allenai/ike/IkeBatchAPI.scala

```diff
@@ -0,0 +1,226 @@
+    *       name = "treatements",
+    *       pattern = "{were given|treated with|received|receiving} {CD|NN|JJ|IN}+"
+    *     }
+    *   ]
+    * }
+    *
+    * IndexWithContent = "RCTs"
+    * -------------------------
+    *
+    * @param configFile The config file that holds your patterns.
+    * @return a Map of (String, SearchRequest) where the key (the String) is the name from the config file
+    */
+  def loadPatternsFromConfig(configFile: String): Map[String, SearchRequest] = {
+    val patternConfig = ConfigFactory.load(configFile)
+    val hugeLimit = 100000000
+    val extractionPatterns = patternConfig.getConfigList("ExtractionPatterns.patterns").asScala.map { config =>
+      config.getString("name") -> SearchRequest(Left(config.getString("pattern")), None, None, SearchConfig(hugeLimit, 1))
+    }.toMap
+    extractionPatterns
+  }
+
+  /** Peforms the search over the corpus given the specific pattern to look for (which was loaded from config)
+    *
+    *   // TODO: Should probably refactor to reuse parts of IKEToolWebApp, maybe some common pieces for both?
+    *   // TODO: lots of copypasta from IKEToolWebApp, gross
+    *
+    * @param patternName The pattern you are looking for
+    * @param req The SearchRequest object
+    * @param indexWithContent This is the name of hte index that holds hte content you care about. You need it
+    *                         because you want the filename and the sentence/content for this pattern's match
+    * @return A Seq of ApiExtractionResult case classes which hold the results
+    */
+  def doSearch(patternName: String, req: SearchRequest, indexWithContent: String): Seq[ApiExtractionResult] = {
+    import org.allenai.ike.JsonSerialization._
+
+    val query = SearchApp.parse(req).get
+
+    val (tables, patterns) = req.userEmail match {
+      case Some(userEmail) => (
+        Tablestore.tables(userEmail),
+        Tablestore.namedPatterns(userEmail)
+      )
+      case None => (Map.empty[String, Table], Map.empty[String, NamedPattern])
+    }
+
+    val interpolatedQuery = QueryLanguage.interpolateQuery(
+      query,
+      tables,
+      patterns,
+      word2vecPhrasesSearcher,
+      tableExpander
+    ).get
+
+    val foundResults = searchApps.map { case (name, searcher) => searcher.search(interpolatedQuery, req.config).get }
+    val qexpr = SearchApp.parse(req).get
+
+    // brakes the flatmap if empty
+    if (foundResults.head.isEmpty) {
+      return List[ApiExtractionResult]()
+    }
+
+    val extractions = foundResults.flatMap {
+      case rr =>
+        rr.map {
+          case blab =>
+            val docId = blab.sourceDocument.getOrElse(-1) // TODO: Should barf on -1
+
+            val extractedString = blab.matchWords.mkString(" ")
+
+            val extractionStart = blab.matchOffset.start
+            val extractionEnd = blab.matchOffset.end
+
+            // Lucene document is XML. Assuming one index where this documnent lives for now. TODO: sketchy...
```

Comment:
Ok. So is this a feature for this PR? Honestly, I'm not yet familiar enough with the codebase to do something like that yet. In the meantime, I will check-in what I have since it does address a few of the concerns.